service/src/main/resources/api/service_openapi.yaml

```diff
@@ -598,6 +643,70 @@ components:
       description: Enum representing a cloud platform type.
       enum: [ 'AZURE', 'GCP' ]
 
+    SpendReport:
+      type: object
+      required:
+        - spendDetails
+        - spendSummary
+      properties:
+        spendDetails:
+          description: details of spend data, categorized by given key
+          type: array
+          items:
+            $ref: '#/components/schemas/SpendReportingAggregation'
+        spendSummary:
+          $ref: '#/components/schemas/SpendReportingForDateRange'
+
+    SpendReportingAggregation:
+      type: object
+      required:
+        - aggregationKey
+        - spendData
+      properties:
+        aggregationKey:
+          type: string
+          enum:
+            - Category
+          description: Key indicating how spendData has been aggregated.
+        spendData:
+          type: array
+          items:
+            $ref: '#/components/schemas/SpendReportingForDateRange'
+
+    SpendReportingForDateRange:
+      required:
+        - cost
+        - credits
+        - currency
+        - startTime
+        - endTime
+      type: object
+      properties:
+        cost:
+          type: string
+          description: total cost for date range
+        credits:
+          type: string
+          description: credits towards reported cost. cost - credits = amount billed to users
```

Comment:
Probably worth nothing here that `credits` will always be 0 for Azure billing profiles.

Reply:
Swagger document updated